src/map/status.c

```diff
@@ -11114,18 +11263,192 @@ static int status_change_end_(struct block_list *bl, enum sc_type type, int tid,
+					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_GROUND);
+					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_GROUND);
+					int sc_tck = (int)DIFF_TICK(td->tick, timer->gettick());
+					int sc_ttl = sc->data[SC_ARMOR_RESIST]->total_tick;
+					clif->status_change_sub(bl, sc_icn, sc_typ, 1, sc_tck, sc_ttl, 0, 0, 0);
+					remove_icon = false;
+				}
+
+				// Fire
+				int sum_fire = sc->data[SC_ARMOR_RESIST]->val3;
+				if (sc->data[SC_RESIST_PROPERTY_FIRE] != NULL)
+					sum_fire += sc->data[SC_RESIST_PROPERTY_FIRE]->val1;
+				if (sc->data[SC_ARMOR_RESIST]->val3 > 0 && sum_fire > 0) {
+					const struct TimerData *td = timer->get(sc->data[SC_ARMOR_RESIST]->timer);
+					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_FIRE);
+					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_FIRE);
+					int sc_tck = (int)DIFF_TICK(td->tick, timer->gettick());
+					int sc_ttl = sc->data[SC_ARMOR_RESIST]->total_tick;
+					clif->status_change_sub(bl, sc_icn, sc_typ, 1, sc_tck, sc_ttl, 0, 0, 0);
+				}
+			}
+
+			break;
+		case SC_RESIST_PROPERTY_FIRE:
+			if (sc->data[SC_ARMOR_RESIST] != NULL) {
+				// Fire
+				int sum_fire = sc->data[SC_ARMOR_RESIST]->val3;
+				if (sc->data[SC_RESIST_PROPERTY_GROUND] != NULL)
+					sum_fire += sc->data[SC_RESIST_PROPERTY_GROUND]->val2;
+				if (sc->data[SC_ARMOR_RESIST]->val3 > 0 && sum_fire > 0) {
+					const struct TimerData *td = timer->get(sc->data[SC_ARMOR_RESIST]->timer);
+					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_FIRE);
+					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_FIRE);
+					int sc_tck = (int)DIFF_TICK(td->tick, timer->gettick());
+					int sc_ttl = sc->data[SC_ARMOR_RESIST]->total_tick;
+					clif->status_change_sub(bl, sc_icn, sc_typ, 1, sc_tck, sc_ttl, 0, 0, 0);
+					remove_icon = false;
+				}
+
+				// Water
+				int sum_water = sc->data[SC_ARMOR_RESIST]->val1;
+				if (sc->data[SC_RESIST_PROPERTY_WATER] != NULL)
+					sum_water += sc->data[SC_RESIST_PROPERTY_WATER]->val1;
+				if (sc->data[SC_ARMOR_RESIST]->val1 > 0 && sum_water > 0) {
+					const struct TimerData *td = timer->get(sc->data[SC_ARMOR_RESIST]->timer);
+					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_WATER);
+					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_WATER);
+					int sc_tck = (int)DIFF_TICK(td->tick, timer->gettick());
+					int sc_ttl = sc->data[SC_ARMOR_RESIST]->total_tick;
+					clif->status_change_sub(bl, sc_icn, sc_typ, 1, sc_tck, sc_ttl, 0, 0, 0);
+				}
+			}
+
+			break;
+		case SC_RESIST_PROPERTY_WIND:
+			if (sc->data[SC_ARMOR_RESIST] != NULL) {
+				// Wind
+				int sum_wind = sc->data[SC_ARMOR_RESIST]->val4;
+				if (sc->data[SC_RESIST_PROPERTY_WATER] != NULL)
+					sum_wind += sc->data[SC_RESIST_PROPERTY_WATER]->val2;
+				if (sc->data[SC_ARMOR_RESIST]->val4 > 0 && sum_wind > 0) {
+					const struct TimerData *td = timer->get(sc->data[SC_ARMOR_RESIST]->timer);
```

Comment:
``timer->get`` may return NULL. also sc->data[SC_ARMOR_RESIST]->timer can be invalid timer. probably need add checks for timer field and return value of ``timer->get``?